### DISCUSSION TITLE
How do I log to a file?

### DISCUSSION THREAD

**Discussion by vamshiaruru-virgodesigns** [apollographql/router#2164] [Q&A] [answered]
Hi! We are trying to switch from apollo federation to apollo router for performance reasons. It looks like router uses env_log crate, which doesn't log to files. What are performance implications of this? Is there a way to perform async logging (or batched logging) using creates like tracing file appender?

Thanks in advance for the help :)

**Comment by bnjjj** [comment, marked as answer]
For now the only way to log to a file is to use unix log redirection

**Reply by dhruvmehtad11** [reply]
@bnjjj can you please check this discussion also [Using basic logging instead of json format logging for unix file redirection](https://github.com/apollographql/router/discussions/2875) .
